src/tr2/game/objects/effects/snow_sprite.c

```diff
@@ -1,6 +1,27 @@
 #include "game/objects/effects/snow_sprite.h"
 
-#include "global/funcs.h"
+#include "game/effects.h"
+#include "game/math.h"
+#include "global/vars.h"
+
+void __cdecl SnowSprite_Control(const int16_t fx_num)
+{
+    FX *const fx = &g_Effects[fx_num];
+    const OBJECT *const obj = Object_GetObject(fx->object_id);
+
+    fx->frame_num--;
+    if (fx->frame_num > obj->mesh_count) {
```

Comment:
I think it's `<=` similar to other sprite effects, all sprite sequence lengths are negative.

Reply:
Thanks – addressed.